torch/_refs/__init__.py

```diff
@@ -2393,16 +2383,26 @@ def mean(
 @register_decomposition(aten.std_mean.correction)
 def std_mean(
     a: TensorLikeType,
-    dim: Union[Optional[int], Optional[List[int]]] = None,
+    dim: Optional[DimsType] = None,
     *,
     unbiased: Optional[bool] = None,
     keepdim: bool = False,
     correction: Optional[int] = None,
 ):
     dim, unbiased = _dim_var_dispatch(dim, unbiased)
-    s = std(a, dim, unbiased, keepdim, correction=correction)
-    m = mean(a, dim, keepdim)
-    return s, m
+    correction = utils.set_correction(unbiased, correction)
+    opmath_dtype, dtype = utils.reduction_dtypes(
+        a, REDUCTION_OUTPUT_TYPE_KIND.COMPLEX_TO_FLOAT
+    )
+    original_dtype = a.dtype
+    a = _maybe_convert_to_dtype(a, opmath_dtype)
```

Comment:
Shouldn't all this conversion stuff be done in `var_mean`? Isn't it alright here to just call `var_mean` and take the `sqrt` of the result? Same above.

Reply:
No, the sqrt must be done in the opmath type or we lose precision and effectively half the dynamic range of float16. Most references have similar  promotion logic but it's handled inside of a wrapper.

Reply:
ugh, yeah, I was hoping this were not the case. Fair enough